daemon/src/model/cfd.rs

```diff
@@ -774,19 +775,19 @@ impl Cfd {
         Ok(self.event(event))
     }
 
-    pub fn roll_over(self, rollover_completed: RolloverCompleted) -> Result<Event> {
+    pub fn roll_over(self, completed: RolloverCompleted) -> Result<Event> {
         // TODO: Compare that the version that we started the rollover with is the same as the
         // version now. For that to work we should pass the version into the state machine
         // that will handle rollover and the pass it back in here for comparison.
         // Daniel says: I don't think the version check will help in addition to what we check in
         // can_rollover. What difference would it make? If we are completed rollover and are
         // still able to apply it we should, otherwise we might produce inconsistent state with the
         // other party.
-        if let Err(e) = self.can_rollover() {
+        if let (Err(e), RolloverCompleted::Succeeded { .. }) = (self.can_rollover(), &completed) {
```

Comment:
This was interesting. I noticed that we were hitting this `bail` when receiving a `RollOverCompleted::Failed` caused by failing on this very `can_rollover` check inside of a different method.

Reply:
Oh my, yes! No checks needed for failed/rejected. Good catch!